src/proteus/dswx_hls.py

```diff
@@ -3224,11 +3233,17 @@ def _warp(input_file, geotransform, projection,
                            tile_max_x_utm + margin_m,
                            tile_srs, file_srs)
 
-    # if it doesn't cross the antimeridian, reproject
-    # input file using gdalwarp
-    if not (file_srs.IsGeographic() and tile_min_x < 180 and
-            tile_max_x >= 180):
-
+    # The dateline crossing by the tile is tested by the condition:
+    #     `tile_min_x < 180 and tile_max_x >= 180`
+    # The ancillary input only requires reprojection if it is in
+    # geographic coordinates and if its longitude domain is represented
+    # within the [-180, +180] range, verified by the test `min_x < -170`,
```

Comment:
What prompted the decision to use -170 ? Could you please add a comment in the code explaining this? Thanks!

Reply:
There's no specific reason to be `-170`. It could be `-160` or even `0`. The important thing is to know that whether the input is using the longitude range [-180, 180] (but obviously we need some margin to test, we cannot use `-180`).